include/deal.II/grid/filtered_iterator.h

```diff
@@ -753,6 +773,122 @@ make_filtered_iterator (const BaseIterator &i,
 
 
 
+#ifdef DEAL_II_WITH_CXX11
+namespace internal
+{
+  namespace FilteredIterator
+  {
+    template <typename BaseIterator, typename TypeList>
```

Comment:
Please add a class in front here that says that this creates a nested sequence of `FilteredIterator<FilteredIterator<...<BaseIterator>...>>` with as many levels of `FilteredIterator` classes as there are elements in the `TypeList` if the latter is given as a `std::tuple<Args...>`.